agent/api/ecsclient/client.go

```diff
@@ -154,16 +154,43 @@ func (client *APIECSClient) registerContainerInstance(clusterRef string, contain
 	// Add additional attributes such as the os type
 	registrationAttributes = append(registrationAttributes, client.getAdditionalAttributes()...)
 	registerRequest.Attributes = registrationAttributes
+	registerRequest = client.setInstanceIdentity(registerRequest)
+
+	resources, err := client.getResources()
+	if err != nil {
+		return "", err
+	}
+
+	registerRequest.TotalResources = resources
+	resp, err := client.standardClient.RegisterContainerInstance(&registerRequest)
+	if err != nil {
+		seelog.Errorf("Unable to register as a container instance with ECS: %v", err)
+		return "", err
+	}
+	seelog.Info("Registered container instance with cluster!")
+	err = validateRegisteredAttributes(registerRequest.Attributes, resp.ContainerInstance.Attributes)
+	return aws.StringValue(resp.ContainerInstance.ContainerInstanceArn), err
+}
+
+func (client *APIECSClient) setInstanceIdentity(registerRequest ecs.RegisterContainerInstanceInput) ecs.RegisterContainerInstanceInput {
```

Comment:
The variables in this function seem unnecessarily complicated.

How about instead of declaring two variables instanceIdentityDoc and Signature, we just set registerRequest.Document and registerRequest.Signature to "" in the beginning. Remove lines 181, 182. In line 187 and 195, assign directly to registerRequest. Would that be simpler?